### ISSUE TITLE
When a movie/tv show is deleted the subtitles remain

### ISSUE THREAD

**Issue by taimoorq** [pannal/Sub-Zero.bundle#152]
Thanks for a great plugin. One question i have though is whether there's a way to delete subtitles for media that has been deleted? For tv shows this causes issues because even though the show is deleted the subtitles remain in the folder which causes the left over shows to pick up the wrong subtitles.

**Comment by pannal** [comment]
I get that there should be some cleaning after TV shows are deleted.

But how does an old TV show pick up old subtitles of a deleted TV show? Are you using metadata storage?

**Comment by ukdtom** [comment]
OP is storing subs next to the media, as sidecars.
When PMS allows the King to delete a media, PMS will do so, but it'll not touch files put in by 3.Party.

IMHO, this should be handled by Plex, where they should nuke everything file that is part of that media, according to the database.

And this is not related to Sub-Zero, but also goes for manually added subs

/T

**Comment by taimoorq** [comment]
ukdtom is correct the subtitles get placed in the same folder as the media. is there a way to change that? Would that make a difference?

As for the subtitles being picked up by other shows, not sure how that happens. When i go to the available subtitles i see them all listed with the same name and can change the active subtitle until i get to the right one.

Do you guys not experience this in your setup? What settings could i have that are different. I didn't do any additional configuration post install so i think i am going with the default settings.

**Comment by ukdtom** [comment]
Haven't tested, but would guess, that if subs fetched by Sub-Zero was stored in the bundle directory (Meta) then they would be cleaned out during a delete

Check Sub-Zero settings:

![storewithmeta](https://cloud.githubusercontent.com/assets/4699449/15268409/eb768796-19dd-11e6-8995-a2e55fea2055.png)

/Tommy

**Comment by taimoorq** [comment]
Thanks Tommy. I'll try that out and see if that makes a difference.

**Comment by pannal** [comment]
Switching to internal storage is a solution, but not a very elegant one.

Adding a cleanup task which runs fairly rare, like once every day, would be a nice feature addition. I don't know how to implement this from the get-go, though, I'll have to think on that one, first.

**Comment by ukdtom** [comment]
Take care with the size of the dict!
Some people out there does literally have thousands of medias :-)

And as said before, I really think this is a PMS issue, and not a SZ issue

/T

**Comment by pannal** [comment]
I'm not too sure about the Dict size and issues regarding that.
As far as I can tell, the PMS saves it to the internal meta storage whatever it takes (correct me if I'm wrong). We're talking about kilobytes of information storage, if they use JSON (or even pickle) serialization.

**Comment by ukdtom** [comment]
Nope regarding the dict size.
Was simply raising a flag here :-)

And regarding direct access to the database, then that's a No Go, since Plex sadly didn't provide the _sqlite3.so file for all platforms, meaning no way to import SQLite :-(
(All NAS platforms hit by this, as well as a lot of Linux ones as well)

And when talking to Plex, this was considered as "Will not fix"

/T

**Comment by pannal** [comment]
I'm with you there. Using sqlite with plex *is* possible, but only in a hacky and very hard to support way.

I'd go for a different (not python-native-sqlite) way if need be.

We should raise a very different issue with the Plex guys in the long term: 

Urging them to ditch the stupid restricted python "implementation". Python is never wholly and securely sandboxed, just because of the way it is, and RestrictedPython (their "plugin implementation") has been broken from day one.

I've never actively or passively been held-back by useless "security-concerns" as much as I've been when programming something, as with the Plex environment. And for no reason - look at my [basic approach of re-enabling certain core features](https://github.com/pannal/Sub-Zero.bundle/blob/master/Contents/Libraries/Shared/subzero/sandbox.py#L6) - they're simply holding us back.

**Comment by pannal** [comment]
@taimoorq is this ticket still relevant? I've added automatic leftover subtitle cleanup some versions ago. Did that fix your issue?

**Comment by pannal** [comment]
I guess it is. [See here](https://forums.plex.tv/discussion/comment/1364449/#Comment_1364449).

**Comment by taimoorq** [comment]
yeah feel free to close this one. I'm good now.

Thanks!

**Comment by pannal** [comment]
Hmm no, this isn't totally resolved. I've fixed leftover subtitles but not after-deleted-leftovers.

**Comment by ukdtom** [comment]
If SZ was running as a deamon, you could maybe use WebHooks ?

**Comment by pannal** [comment]
Yeah, maybe. I've implemented the websocket stuff of plex_activity in the last release - maybe they expose the delete event.
